### PR TITLE
[Android] Sends cookies with resource requests

### PR DESCRIPTION
**Pull request by 1rahul1**
To attach cookie with all resource requests for the page loaded
# Summary
* This PR solves this feature request [#1532 ](https://github.com/react-native-community/react-native-webview/issues/1532)

* How did you implement the solution?
I implemented this solution by adding a map for storing cookies for authority name and overriding the shouldInterceptRequest callback in RNCWebViewClient.

* What areas of the library does it impact?
RNCWebViewManager, RNCWebViewClient

## Compatibility

| OS      | Implemented |
| Android |✅|

## Checklist

- [x] I have tested this on a device and a simulator
- [ ] I added the documentation in `README.md`
- [ ] I mentioned this change in `CHANGELOG.md`
- [ ] I updated the typed files (TS and Flow)
- [ ] I added a sample use of the API in the example project (`example/App.js`)

### CODE REVIEW

**Comment by jamonholmgren** [comment]
Thanks for submitting this, @1rahul1 ! Does this only affect Lollipop?

**Comment by TommyLeong** [comment]
@jamonholmgren  Unfortunately I wasn't having issue with Android, earlier my issue of passing cookie with resource request was with iOS. Somehow the current version of Android works fine for me for this part.

**Comment by gkode** [comment]
@jamonholmgren @jamonholmgren @TommyLeong  I've faced this issue and customize it locally.  I have tested this code and it works fine. Can we merge it?

**Comment by gkode** [comment]
@Titozzz

File: android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java [hunk 1]
```diff
@@ -799,6 +801,20 @@ public boolean shouldOverrideUrlLoading(WebView view, WebResourceRequest request
       return this.shouldOverrideUrlLoading(view, url);
     }
 
+    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+    @Override
+    public WebResourceResponse shouldInterceptRequest(WebView view, WebResourceRequest request) {
+      String cookie = request.getRequestHeaders() == null ? null : request.getRequestHeaders().get("cookie");
```
Comment (dwxw) [inline android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java:807]: Hi, just wondering if "cookie" should be case insensitive?  It didn't work when I declared my headers as `headers={{ Cookie: "key=value" }}`.  Appears to be case insensitive on iOS.

Comment (1rahul1) [reply android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java:807]: Hi @dwxw, Did you check this in Android? Did you check `headers={{ cookie: "key=value" }}`?

Comment (dwxw) [reply android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java:807]: Yes I tried your change and compared it to the existing iOS behaviour.  The headers prop works with `Cookie` or `cookie` in iOS, but with this change only `cookie` works on Android.  [AFAIK](https://stackoverflow.com/questions/5258977/are-http-headers-case-sensitive/5259004#5259004) the http header should be case-insensitive so you should handle both cases.

Comment (jamonholmgren) [reply android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java:831]: @1rahul1 Do you agree with @dwxw? If you want to make this case insensitive, I'll merge & release after. Or if not, I can merge and release now.

**Comment by jamonholmgren** [comment]
We do intend to merge & release this. Just one outstanding question.

File: android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java (hunk 1 above)
Comment (1rahul1) [reply android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java:831]: @dwxw, thanks for checking that. I checked it on Android 9.0 but wasn't able to reproduce the issue which you pointed out. For me, it worked for both cases. Can you give more details on which version of Android used and how to reproduce it?

Comment (1rahul1) [reply android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java:831]: @jamonholmgren I checked this and it worked for me for both the cases. In the headers from [ `request.getRequestHeaders()`](https://github.com/1rahul1/react-native-webview/blob/9970504f2a6601531cc8557bd82f4342f9c1b3f2/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java#L831), the **keys** of the headers are in **lowercase** for the headers passed in the source prop, **irrespective** of the case of the key passed. [Here](https://github.com/1rahul1/react-native-webview/blob/9970504f2a6601531cc8557bd82f4342f9c1b3f2/android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java#L831), I have used "cookie" which is in lowercase. So, I think the case-insensitive case is already handled.

Comment (dwxw) [reply android/src/main/java/com/reactnativecommunity/webview/RNCWebViewManager.java:831]: I've had another look and I still have this problem where only "cookie" works for me. The server does not receive the cookie unless I use "cookie", but after digging into this code it must be something else.  I've looked at `request.getRequestHeaders()` myself and it always returns a lowercase "cookie" no matter the case I use in the `headers` prop.  So not sure what my problem is but I think @1rahul1's code looks good.  Sorry for any confusion.

**Comment by gkode** [comment]
@jamonholmgren Can you please merge it

**Comment by gkode** [comment]
@jamonholmgren  Can you please specify what's your "outstanding question" on this PR?

**Comment by Zaidos9** [comment]
Don't have access to resolve conflict. Created [PR 1700 [Android] Sends cookies with resource requests](https://github.com/react-native-webview/react-native-webview/pull/1700) to replace this PR.

**Pull request closed without merging**